### PR TITLE
[NFC][Concurrency] Add a few test cases for isolated parameters.

### LINKED ISSUE #62568: `Task` created from a method with an isolated parameter breaks actor isolation
**Description**
A `Task` created from a [method that executes within the isolated context of a passed-in actor](https://github.com/apple/swift-evolution/blob/main/proposals/0313-actor-isolation-control.md#actor-isolated-parameters) should inherit the execution context of the passed-in actor, but does not.

**Steps to reproduce**
The following code fails to compile:
```swift
func execute<ActorType: Actor>(
    on isolatedActor: isolated ActorType,
    task: @Sendable (isolated ActorType) -> Void)
{
    // Compiler correctly allows this task to execute synchronously.
    task(isolatedActor)
    // Start a task that inherits the current execution context (i.e. that of the isolatedActor)
    Task {
        // BUG: compiler insists that we must `await` the `task`, meaning that this Task is not executing in the isolatedActor's execution context
        task(isolatedActor) // Compiler error: Expression is 'async' but is not marked with 'await'
    }
}
```
Here's a screenshot of the same code with the inlined compiler error for folk who like Xcode's syntax hilighting:

A screenshot of the code above in Xcode

**Expected behavior**
I would expect the above code to compile without modification.

Per [the Task documentation](https://developer.apple.com/documentation/swift/task/init(priority:operation:)-5ltye), a not-detached `Task` "inherits the priority and actor context of the caller", meaning that a `Task` created in an `isolated ActorType` method should execute within the isolated context of the passed-in actor.

**Environment**
- Swift compiler version info

I was able to reproduce this issue on the following swift compiler versions (I did not try other versions):
```
swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

```
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

- Xcode version info

I was able to reproduce this issue on the following Xcode versions (I did not try other versions):
```
Xcode 14.1
Build version 14B47b
```

```
Xcode 14.2
Build version 14C18
```

- Deployment target:

```
.macOS(.v10_15),
.iOS(.v13),
.tvOS(.v13),
.watchOS(.v6),
```
**Additional context**

This bug report was first posted [as a question in the Swift forums](https://forums.swift.org/t/should-task-inherit-the-execution-context-of-an-isolated-parameter/61999/1).

Unlike #62503 and #62505, I have not found a way to force the compiler to execute the `Task` above in the expected execution context.

### PR DESCRIPTION
**Pull request by hborla**
Adds a few test cases for isolated parameter issues that are now fixed on `main`:
* Inheriting actor context within a function that uses an isolated parameter.
* A non-escaping, synchronous closure used within a function with an isolated parameter.

Resolves https://github.com/apple/swift/issues/62568, rdar://90849047

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please smoke test

File: test/Concurrency/isolated_parameters.swift [hunk 1]
```diff
@@ -319,3 +319,32 @@ func isolatedClosures() {
...
   }
 }
+
+// Test case for https://github.com/apple/swift/issues/62568
+func execute<ActorType: Actor>(
+  on isolatedActor: isolated ActorType,
+  task: @escaping @Sendable (isolated ActorType) -> Void)
+{
+  // Compiler correctly allows this task to execute synchronously.
+  task(isolatedActor)
```
Comment (ktoso) [inline test/Concurrency/isolated_parameters.swift:329]: a bit confusing to call this "task" but code is good 👍

**Review by ktoso — APPROVED** [review]
LGTM!

File: test/Concurrency/isolated_parameters.swift (hunk 1 above)
Comment (hborla) [reply test/Concurrency/isolated_parameters.swift:329]: Fair, but I copied the test case from https://github.com/apple/swift/issues/62568

**Pull request merged**